src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -2167,7 +2167,8 @@ else if (null != sourceCryptoMeta) {
                         if (bulkNullable) {
                             tdsWriter.writeByte((byte) 0x08);
                         }
-                        tdsWriter.writeDouble((float) colValue);
+                        float floatValue = ((Number)colValue).floatValue();
+                        tdsWriter.writeDouble(floatValue);
```

Comment:
Can you make that change you suggested? While it is cleaner, it doesn't make sense to define a variable that is only used on the immediate following line.
```suggestion
                        tdsWriter.writeDouble(((Number)colValue).floatValue());
```